app/frontend/src/features/donations/DonationsBox.tsx

```diff
@@ -0,0 +1,495 @@
+import {
+  Divider,
+  FormControl,
+  FormControlLabel,
+  FormHelperText,
+  makeStyles,
+  Radio,
+  RadioGroup,
+  Typography,
+} from "@material-ui/core";
+import classNames from "classnames";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import {
+  DONATIONSBOX_ALERT_SUCCESS,
+  DONATIONSBOX_ALERT_WARNING,
+  DONATIONSBOX_CURRENCY,
+  DONATIONSBOX_MONTHLY,
+  DONATIONSBOX_NEXT,
+  DONATIONSBOX_ONETIME,
+  DONATIONSBOX_RECURRING,
+  DONATIONSBOX_RECURRING_ARIA,
+  DONATIONSBOX_REQUIRED,
+  DONATIONSBOX_TEXT,
+  DONATIONSBOX_TITLE,
+  DONATIONSBOX_VALUES,
+} from "features/donations/constants";
+import { Error as GrpcError } from "grpc-web";
+import { useMemo, useRef, useState } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { useMutation } from "react-query";
+import { useLocation } from "react-router-dom";
+import { service } from "service";
+
+const useStyles = makeStyles((theme) => ({
+  donationsBox: {
+    padding: theme.spacing(2),
+    border: `2px solid ${theme.palette.grey[200]}`,
+    borderRadius: theme.shape.borderRadius * 2,
+  },
+
+  donationsBoxRow: {
+    gridTemplateColumns: "repeat( auto-fit, minmax(160px, 1fr) )",
+    gridAutoRows: "2.75rem",
+    display: "grid",
+    gridGap: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+  },
+
+  donationsBoxSubRow: {
+    gridTemplateColumns: "repeat( auto-fit, minmax(72px, 1fr) )",
+    display: "grid",
+    gridGap: theme.spacing(2),
+  },
+
+  buttonSecondary: {
+    "&&": {
+      boxShadow: "initial",
+      color: theme.palette.grey[600],
+      cursor: "pointer",
+      fontWeight: 700,
+      fontSize: theme.typography.button.fontSize,
+      transition: `color ${theme.transitions.duration.short}ms ${theme.transitions.easing.easeInOut}`,
+      justifyContent: "center",
+      alignItems: "center",
+      border: `2px solid ${theme.palette.grey[200]}`,
+      borderRadius: "0.5rem",
+      backgroundColor: theme.palette.grey[200],
+      margin: "initial",
+      height: "100%",
+      width: "100%",
+      display: "flex",
+    },
+
+    "&:hover": {
+      border: `2px solid ${theme.palette.primary.main}`,
+      backgroundColor: theme.palette.background.paper,
+      color: theme.palette.primary.main,
+      transition: `color ${theme.transitions.duration.short}ms ${theme.transitions.easing.easeInOut}`,
+    },
+  },
+
+  buttonSecondaryRadio: {
+    "&&": {
+      boxShadow: "initial",
+      margin: 0,
+    },
+    "&:hover > .MuiFormControlLabel-label": {
+      border: `2px solid ${theme.palette.primary.main}`,
+      backgroundColor: theme.palette.background.paper,
+      color: theme.palette.primary.main,
+      transition: `color ${theme.transitions.duration.short}ms ${theme.transitions.easing.easeInOut}`,
+    },
+    "& > .MuiRadio-root": {
+      position: "absolute",
+      left: "-10000px",
+    },
```

Comment:
> https://webaim.org/techniques/css/invisiblecontent/ suggests position: absolute; left: -10000px; so I will change it to that. It's probably 10ns more performant than 0 opacity.

I was less concerned about the performance, but more that this removes the focus trap state for screen readers, which the `position: "fixed"; opacity: 0;` combo still retain (shows a box around an invisible thing, but I think still better than nothing) without disrupting the text that's inside the button.

Also we should add some focus states (the blue outline) on these "radio buttons" like what we got on the $ buttons. Since they are fake buttons but styled to look like one, currently it doesn't show any which makes them pretty unusable with keyboard only at the moment without a lot of guess work.

Reply:
If having "fake" radio buttons is a problem, I can refactor and make them "normal" buttons. What I don't see is to have an invisible radio button just over the fake button, as this can confuse users in some edge situations. Also, they could be moved unintentionally leading to a situation where the radio button could be placed anywhere else, causing people clicking them when wanting to click another item. This in a "money page" is a situation we should avoid, imo.

Reply:
> Also, they could be moved unintentionally leading to a situation where the radio button could be placed anywhere else, causing people clicking them when wanting to click another item.

It's inside the button at the moment anyway and since it's relative to the body and out of normal flow, I don't see how that will change and hence clicking the radio button outside of the rectangular button itself is a moot point here.

Anyway, I think a "normal" button is probably better since it gives us the same focus styling when tabbing with keyboard like on the dollar buttons for free, so if that's easier I'd go with that. The radio button is a clever solution if we were sticking to using the normal radio buttons but clearly not the case here with the design...

Reply:
Ok, I'll take care of that. This weekend I will not be able to work in this, so I'd suggest to merge this in order to have the functionality live, and then I'll make a commit with the refactor.